File: Code/isPalindrome_q9.js
/*
Given an integer x, return true if x is a palindrome, and false otherwise.
*/

function isPalindrome(num) {
    let str = num.toString()
    let split = str.split("")
    let reverse = split.reverse()
    let reversed = reverse.join("")

    let isPalindrome = false

    if (str == reversed) {
        isPalindrome = true
    }

    return isPalindrome
}

console.log(isPalindrome(44344))
console.log(isPalindrome(121))

File: Code/isPalindrome_q9.py
"""
Given an integer x, return true if x is a palindrome, and false otherwise.
"""
"""
    - First thought, convert to string, reverse, compare
    - Should set variable to False, only change if reversed string equals original string
"""


def isPalindrome(num):
    str_num = str(num)  # convert int to string
    reverse_str = str_num[::-1]  # reverses string for comparison

    isPalindrome = False

    if str_num == reverse_str:
        isPalindrome = True

    return isPalindrome


print(isPalindrome(121))
print(isPalindrome(991))
